### PR TITLE
UICHKIN-283: UI tests replacement with RTL/Jest for component Multipi…

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
UI tests replacement with RTL/Jest for component `MultipieceModal`

## Refs
https://issues.folio.org/browse/UICHKIN-283

## Screenshots
Screen Shot 2022-04-12 at 1 21 47 PM

### CODE REVIEW

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  fireEvent,
+  render,
+  screen,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+import {
+  componentPropsCheck,
+  getById,
+} from '../../../test/jest/helpers'
```
Comment (Stanislau-Snapok) [inline src/components/MultipieceModal/MultipieceModal.test.js:12]: Could you please add semicolon at the end?
Also could you please move this import after stripes components?

File: src/components/MultipieceModal/MultipieceModal.js [hunk 2]
```diff
@@ -12,19 +15,26 @@ import {
 } from '@folio/stripes/components';
 
 const MultipieceModal = (props) => {
-  const { item, onClose, onConfirm } = props;
+  const {
+    intl: { formatMessage },
```
Comment (Stanislau-Snapok) [inline src/components/MultipieceModal/MultipieceModal.js:19]: Could you please add `intl` in `propTypes`?

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('with default props', () => {
+    beforeEach(() => {
+      render(<MultipieceModal {...testDefaultProps} />);
+    });
+
+    it('should render multipiece Modal component', () => {
```
Comment (Stanislau-Snapok) [inline src/components/MultipieceModal/MultipieceModal.test.js:70]: Could you please wrap component names in double quotes?
It relates for whole test I think.

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalConfirmButton));
+
+      expect(testOnConfirm).toHaveBeenCalled();
+    });
+
+    it('should render multipiece modal cancel Button component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalCancelButton, {}, true);
+    });
+
+    it('should render multipiece multipiece modal cancel Button component label', () => {
```
Comment (Stanislau-Snapok) [inline src/components/MultipieceModal/MultipieceModal.test.js:105]: Additional "multipiece" I think.

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+    it('should call onClose callback when click on cancel button', () => {
+      expect(testOnClose).not.toBeCalled();
+
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalCancelButton));
+
+      expect(testOnClose).toHaveBeenCalled();
+    });
+
+    it('should render multipiece multipiece modal message label', () => {
```
Comment (Stanislau-Snapok) [inline src/components/MultipieceModal/MultipieceModal.test.js:117]: The same as above.

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 5 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: done. thanks

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 4 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: done. thanks

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 3 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: done

File: src/components/MultipieceModal/MultipieceModal.js (hunk 2 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.js:19]: done

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 1 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: done. thanks

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+      componentPropsCheck(Modal, testIds.multipieceModal, {
+        id: 'multipiece-modal',
+        size: 'small',
+        dismissible: true,
+        label: labelIds.multipieceModalLabel,
+      }, true);
+    });
+
+    it('should render multipiece "ModalFooter" component', () => {
+      componentPropsCheck(ModalFooter, testIds.multipieceModalFooter, {}, true);
```
Comment (artem-blazhko) [inline src/components/MultipieceModal/MultipieceModal.test.js:81]: Does it have any sense to check props if you provide an empty object as expected props?
In this case I would say much better just check that `ModalFooter` was called.

`expect(ModalFooter).toBeCalled();`

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    });
+
+    it('should render multipiece modal confirm "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalConfirmButton, {
+        buttonStyle: 'primary',
```
Comment (artem-blazhko) [inline src/components/MultipieceModal/MultipieceModal.test.js:86]:
```suggestion
        buttonStyle: 'primary',
        onClick: expect.any(Function),
```

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+    it('should call onConfirm callback when click on confirm button', () => {
+      expect(testOnConfirm).not.toBeCalled();
+
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalConfirmButton));
+
+      expect(testOnConfirm).toHaveBeenCalled();
+    });
+
+    it('should render multipiece modal cancel "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalCancelButton, {}, true);
```
Comment (artem-blazhko) [inline src/components/MultipieceModal/MultipieceModal.test.js:103]: The same with props checking here

File: src/components/MultipieceModal/MultipieceModal.test.js [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    });
+
+    it('should render multipiece modal message label', () => {
+      expect(screen.getByText(labelIds.multipieceModalMessage)).toBeVisible();
+    });
+
+    it('should render number of pieces "KeyValue" component', () => {
+      componentPropsCheck(KeyValue, testIds.numberOfPieces, {
+        label: labelIds.itemNumberOfPieces,
+        value: '-',
```
Comment (artem-blazhko) [inline src/components/MultipieceModal/MultipieceModal.test.js:125]: Could you add `value` variable?
I see that we have 4 cases when `value` equal `-`

Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: Done

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 8 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js]: We have multiple places where Button component is used. We want to check that specific Button(in this case cancel button) is called

File: src/components/MultipieceModal/MultipieceModal.test.js (hunk 7 above)
Comment (pavelspichonak) [reply src/components/MultipieceModal/MultipieceModal.test.js:87]: done

**Review by artem-blazhko — APPROVED** [review]

**Pull request merged**